plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * @package plugins.microsoftTeamsDropFolder
+ * @subpackage batch
+ */
+class KMicrosoftGraphClient
+{
+	const AUTH_URL = 'https://login.microsoftonline.com/{tenantId}/oauth2/token';
+
+	public $apiUrl;
+
+	public $tenantId;
+
+	public $clientSecret;
+
+	public $clientId;
+
+	public $bearerToken;
+
+	public $bearerTokenExpiry;
+
+	function __construct($tenantId, $apiUrl, $clientId, $clientSecret)
+	{
+		$this->apiUrl = $apiUrl;
+		$this->tenantId = $tenantId;
+		$this->clientId = $clientId;
+		$this->clientSecret = $clientSecret;
+	}
+
+	public function authenticate ()
+	{
+		$url = str_replace('{tenantId}', $this->tenantId, self::AUTH_URL);
+
+		$ch = curl_init($url);
+		curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
+
+		$fields = array('grant_type' => 'client_credentials', 'client_id' => $this->clientId, 'client_secret' => $this->clientSecret, 'resource' => $this->apiUrl);
+		curl_setopt($ch, CURLOPT_POSTFIELDS, $fields);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+		$response = json_decode(curl_exec($ch), true);
+		curl_close($ch);
+
+		KalturaLog::info('Auth token generated: [' . $response['access_token'] . '], expiry: ' . date('c', $response['expires_on']));
+		$this->bearerToken = $response['access_token'];
+		$this->bearerTokenExpiry = $response['expires_on'];
+	}
+
+	public function getCallRecord($callRecordId)
+	{
+		$service = $this->apiUrl . "/v1.0/communications/callRecords/$callRecordId";
+		return $this->sendGraphRequest($service);
+	}
+
+	public function getUser($userId)
+	{
+		$service = $this->apiUrl . "/v1.0/users/$userId";
+		return $this->sendGraphRequest($service);
+	}
+
+	public function getDriveItem($driveId, $driveItemId)
+	{
+		$service = $this->apiUrl .  "/v1.0/drives/$driveId/items/$driveItemId";
+		$response = $this->sendGraphRequest($service);
+
+		if ($response)
+		{
+			if (isset ($response['source']) &&
+				$response['source']['application'] == 'Teams' &&
+				isset ($response['media']) && isset ($response['media']['mediaSource'])
+				&& isset ($response['media']['mediaSource']['contentCategory']) && $response['media']['mediaSource']['contentCategory'] == 'Meeting')
```

Comment:
No need to to make all this isset(..) tests, you directly check isset ($response['media']['mediaSource']['contentCategory'], no notice will be raised.